File: openidaas-core/src/main/java/com/qoobot/openidaas/core/domain/Permission.java
package com.qoobot.openidaas.core.domain;

import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.annotation.TableField;
import com.qoobot.openidaas.common.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.Set;

/**
 * 权限领域模型
 *
 * @author QooBot
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("permissions")
public class Permission extends BaseEntity {

    /**
     * 权限编码
     */
    @TableField("perm_code")
    private String permCode;

    /**
     * 权限名称
     */
    @TableField("perm_name")
    private String permName;

    /**
     * 权限类型（menu-菜单，button-按钮，api-API接口）
     */
    @TableField("perm_type")
    private String permType;

    /**
     * 权限路径（用于前端路由或API路径）
     */
    @TableField("path")
    private String path;

    /**
     * 请求方法（GET,POST,PUT,DELETE等）
     */
    @TableField("method")
    private String method;

    /**
     * 权限描述
     */
    @TableField("description")
    private String description;

    /**
     * 父权限ID
     */
    @TableField("parent_id")
    private Long parentId;

    /**
     * 图标
     */
    @TableField("icon")
    private String icon;

    /**
     * 组件路径（前端组件路径）
     */
    @TableField("component")
    private String component;

    /**
     * 是否外链
     */
    @TableField("external_link")
    private Boolean externalLink = false;

    /**
     * 是否隐藏
     */
    @TableField("hidden")
    private Boolean hidden = false;

    /**
     * 排序
     */
    @TableField("sort_order")
    private Integer sortOrder = 0;

    /**
     * 是否启用
     */
    @TableField("enabled")
    private Boolean enabled = true;

    /**
     * 拥有此权限的角色集合
     */
    @TableField(exist = false)
    private Set<Role> roles;

    /**
     * 拥有此权限的用户集合
     */
    @TableField(exist = false)
    private Set<User> users;

    /**
     * 子权限集合
     */
    @TableField(exist = false)
    private Set<Permission> children;

    /**
     * 父权限
     */
    @TableField(exist = false)
    private Permission parent;

    /**
     * 关联的菜单
     */
    @TableField(exist = false)
    private Menu menu;

    /**
     * 权限所属租户
     */
    @TableField(exist = false)
    private Tenant tenant;
}

File: openidaas-admin-ui/src/api/permission.ts
import { http } from '@/utils/http'

/**
 * 权限创建DTO
 */
export interface PermissionCreateDTO {
  permCode: string
  permName: string
  permType: string
  parentId?: number
  path?: string
  component?: string
  icon?: string
  sortOrder?: number
  description?: string
}

/**
 * 权限更新DTO
 */
export interface PermissionUpdateDTO {
  id: number
  permCode?: string
  permName?: string
  permType?: string
  parentId?: number
  path?: string
  component?: string
  icon?: string
  sortOrder?: number
  description?: string
}

/**
 * 权限VO
 */
export interface PermissionVO {
  id: number
  permCode: string
  permName: string
  permType: string
  permTypeDesc?: string
  parentId: number
  path?: string
  component?: string
  icon?: string
  sortOrder: number
  description?: string
  createdAt?: string
  updatedAt?: string
  tenantId?: number
  children?: PermissionVO[]
}

/**
 * 权限API
 */
export const permissionApi = {
  /**
   * 获取所有权限
   */
  getAllPermissions: () =>
    http.get<PermissionVO[]>('/api/permissions'),

  /**
   * 获取权限树
   */
  getPermissionTree: (parentId?: number) =>
    http.get<PermissionVO[]>('/api/permissions/tree', { params: { parentId } }),

  /**
   * 创建权限
   */
  createPermission: (data: PermissionCreateDTO) =>
    http.post<PermissionVO>('/api/permissions', data),

  /**
   * 更新权限
   */
  updatePermission: (data: PermissionUpdateDTO) =>
    http.put<PermissionVO>('/api/permissions', data),

  /**
   * 删除权限
   */
  deletePermission: (id: number) =>
    http.delete(`/api/permissions/${id}`),

  /**
   * 获取权限详情
   */
  getPermissionById: (id: number) =>
    http.get<PermissionVO>(`/api/permissions/${id}`)
}
